src/cogs/rules.py

```diff
@@ -59,19 +60,79 @@ async def on_member_join(self, member):
 
         current_time = time.time()
 
-        self._recent_joins = [
-            x for x in self._recent_joins
-            if current_time - x["join_time"] <= self.file["massjoin_window"]
-        ]
+        if not self.massjoin_active:
+            self._recent_joins = [
+                x for x in self._recent_joins
+                if current_time - x["join_time"] <= self.file["massjoin_window"]
+            ]
+
+        self._recent_joins.append({"join_time": current_time, "id": member.id,
+            "assumed_bot": (
+                ( member.default_avatar_url == member.avatar_url if self.file["massjoin_default_pfp"] else True )
+                and ( time.time() - member.created_at.timestamp() < self.file["massjoin_min_acc_age_val"]
+                    if self.file["massjoin_min_acc_age"] else True ) )
+            })
```

Comment:
format this, every key on its own line + goddamn that condition make it into a function.

"assumed_bot": is_assumed_bot(member)

Reply:
Should the first key be on it's own line too?

Reply:
yes, the `{` should be next to the `(`, but join_time sh ould be on next line, indented from `self`

Reply:
```py
self._recent_joins.append({
    "join_time": ...,
    "assumed_bot": is_assumed_bot(member),
})```